doc/api/v8.md

```diff
@@ -1051,6 +1051,51 @@ added:
 
 Returns true if the Node.js instance is run to build a snapshot.
 
+### Class: `v8.MemoryProfileWatchdog`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+#### `new MemoryProfileWatchdog([options])`
+
+* `interval` {number} How often the watchdog polls.
+* `maxRss` {number} Node.js will take snapshot when rss is
+  greater than or equal to `maxRss`.
+* `maxUsedHeapSize` {number} Node.js will take snapshot when used heap size is
+  greater than or equal to `maxUsedHeapSize`.
+* `filename` {string} Node.js will write heap snapshot to `filename`.
+
+Creates a new `MemoryProfileWatchdog` object, `maxRss` or `maxUsedHeapSize` must
+be setted. `MemoryProfileWatchdog` will create a thread and poll the memory
+info periodically, when the rss or used heap size is greater than or equal to
+the value you set, Node.js will write a heap snapshot to `filename` then stop
```

Comment:
> Why stop the watchdog in case a snapshot was made?

My idea is we should not take much heap snapshots. So just take one currently. 

> There is no hint to user that this happend so application keeps running without watchdog.

Do you mean we should notify the user by callback or event ?

Reply:
For example. Think about a setup where one has serveral hundreds node apps running distributed across several hosts. Every now and then one node app writes a snapshot but tells noone about this.
I would assume that some functionality in the app takes care about user notification to allow followup tasks like:
* upload dump to some store
* graceful restart the app
* restart the watchdog
* ... whatever is appropriate in case a watchdog fires